src/libraries/System.Private.CoreLib/src/System/Threading/Thread.Unix.cs

```diff
@@ -33,6 +33,10 @@ internal static int PollWasiEventLoopUntilResolved(Task<int> mainTask)
             return mainTask.Result;
         }
 
+        internal static void DispatchWasiEventLoop()
```

Comment:
this should not be necessary anymore

Reply:
I was planning to ask you about this.  `PollWasiEventLoopUntilResolved` seems useful when exporting a `Main` function (or any other function that happens to return `int`), but how would it be used for arbitrary exports which might return other types?  Were you planning to make the return type generic?

Reply:
Good question. I was also thinking about it but then I forgot about it.

Yes, I think we could add another PollWasiEventLoopUntilResolved with generic signature. 

https://github.com/dotnet/runtime/blob/main/src/tests/baseservices/compilerservices/UnsafeAccessors/UnsafeAccessorsTests.Generics.cs